R/git.R

```diff
@@ -323,85 +323,127 @@ git_clean <- function() {
 #' Git/GitHub sitrep
 #'
 #' Get a situation report on your current Git/GitHub status. Useful for
-#' diagnosing problems.
+#' diagnosing problems. The default is to report all values; provide values
+#' for `tool` or `scope` to be more specific.
+#'
+#' @param tool Report for __git__, or __github__
+#' @param scope Report globally for the current __user__, or locally for the
+#'   current __project__
+#'
 #' @export
 #' @examples
 #' \dontrun{
+#' # report all
 #' git_sitrep()
+#'
+#' # report git for current user
+#' git_sitrep("git", "user")
 #' }
-git_sitrep <- function() {
+git_sitrep <- function(tool = c("git", "github"),
+                       scope = c("user", "project")) {
+
+  tool <- rlang::arg_match(tool, multiple = TRUE)
+  scope <- rlang::arg_match(scope, multiple = TRUE)
+
   ui_silence(try(proj_get(), silent = TRUE))
 
   # git (global / user) --------------------------------------------------------
-  hd_line("Git config (global)")
-  kv_line("Name", git_cfg_get("user.name", "global"))
-  kv_line("Email", git_cfg_get("user.email", "global"))
-  kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
-  vaccinated <- git_vaccinated()
-  kv_line("Vaccinated", vaccinated)
-  if (!vaccinated) {
-    ui_info("See {ui_code('?git_vaccinate')} to learn more")
-  }
-  kv_line("Default Git protocol", git_protocol())
   init_default_branch <- git_cfg_get("init.defaultBranch", where = "global")
-  kv_line("Default initial branch name", init_default_branch)
+  if ("git" %in% tool && "user" %in% scope) {
+    cli::cli_h3("Git global (user)")
+    kv_line("Name", git_cfg_get("user.name", "global"))
+    kv_line("Email", git_cfg_get("user.email", "global"))
+    kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
+    vaccinated <- git_vaccinated()
+    kv_line("Vaccinated", vaccinated)
+    if (!vaccinated) {
+      ui_info("See {ui_code('?git_vaccinate')} to learn more")
+    }
+    kv_line("Default Git protocol", git_protocol())
+    kv_line("Default initial branch name", init_default_branch)
+  }
 
   # github (global / user) -----------------------------------------------------
-  hd_line("GitHub")
   default_gh_host <- get_hosturl(default_api_url())
-  kv_line("Default GitHub host", default_gh_host)
-  pat_sitrep(default_gh_host)
+  if ("github" %in% tool && "user" %in% scope) {
+    cli::cli_h3("GitHub user")
+    kv_line("Default GitHub host", default_gh_host)
+    pat_sitrep(default_gh_host)
+  }
 
   # git and github for active project ------------------------------------------
-  hd_line("Git repo for current project")
+  if ("project" %in% scope) {
```

Comment:
I propose an early exit here if `!"project" %in% scope`, which will bring all the remaining code up to the top-level, which will be nice.

Reply:
done in e1c13ef